connexion/lifecycle.py

```diff
@@ -34,4 +34,21 @@ def __init__(self,
         self.mimetype = mimetype
         self.content_type = content_type
         self.body = body
-        self.headers = headers or {}
+        self.headers = ConnexionHeaders(headers)
+
+
+class ConnexionHeaders(object):
```

Comment:
IMO, `ConnexionHeaders` should implement the correct representation, which is definitely not a map of names to values.  list of tuples OR map from names to lists of values both seem acceptable.  Given the relatively short number of headers typically, lists of tuples makes the most sense.

It is the responsibility of the layer that translates to/from the underlying framework to transform the correct representation to whatever the underlying framework expects.